src/rust/ide/view/graph-editor/src/component/node/input/area.rs

```diff
@@ -592,11 +600,16 @@ impl Area {
 
                     let port_shape_hover = port_shape.hover.clone_ref();
                     pointer_style_out   <- mouse_out.map(|_| default());
-                    pointer_style_over  <- map3(&mouse_over,&frp.set_ports_active,&port.tp,
-                        move |_,(_,edge_tp),port_tp| {
+                    pointer_style_over  <- map4
+                        (&mouse_over,&frp.set_ports_active,&port.tp,&frp.editor_mode,
+                        move |_,(_,edge_tp),port_tp,editor_mode| {
                             let tp    = port_tp.as_ref().or_else(||edge_tp.as_ref());
                             let color = tp.map(|tp| type_coloring::compute(tp,&styles));
                             let color = color.unwrap_or(any_type_sel_color);
+                            let color = match editor_mode {
+                                EditorMode::Normal    => color,
+                                EditorMode::Profiling => any_type_sel_color
+                            };
```

Comment:
We do the color computation every time, even if it is discarded afterwards (because we are in Profiling mode). Instead, lets compute the color only when we actually use it.